gpii/node_modules/contextManager/src/ContextManager.js

```diff
@@ -207,23 +200,36 @@ gpii.contextManager.evaluateConditions = function (that, lifecycleManager) {
     }, function () {
         fluid.log(fluid.logLevel.ERROR, "contextManager: Failed to apply newly evaluated conditions");
     });
+    return response;
 };
 
-/*
+/**
  * Function to add the current context to the full payload which is output from the matchmaking process.
  *
- * @param that {Object} - gpii.contextManager object
+ * @param that {Component} - gpii.contextManager component
  * @param fullPayload {Object} - the full structure output from the matchmaking process. NOTE: This object will be modified
+ * @param forceContext {String} [optional] - A context name to be selected, overriding environment conditions
  * @return {Object} - the modified fullPayload, now including activeContextName, activeConfiguration
  */
-gpii.contextManager.addActiveContext = function (that, fullPayload) {
+gpii.contextManager.updateActiveContextName = function (that, fullPayload, forceContext) {
     var matchMakerOutput = fullPayload.matchMakerOutput;
-    var activeContexts = gpii.contextManager.utils.findActiveContexts(fluid.get(that.model, "currentContext"), matchMakerOutput);
-
-    // save highest priority context as the applied one
-    fullPayload.activeContextName = activeContexts[0];
+    var newActiveContext;
+    if (forceContext === undefined) {
+        if (fullPayload.forceContext) { // Don't compute conditions if context has been forced
+            newActiveContext = fullPayload.activeContextName;
+        } else {
+            var activeContexts = gpii.contextManager.utils.findActiveContexts(fluid.get(that.model, "currentContext"), matchMakerOutput);
+            // save highest priority context as the applied one
+            newActiveContext = activeContexts[0];
+        }
+    } else {
+        newActiveContext = forceContext;
+        fullPayload.forceContext = true;
```

Comment:
it's a bit confusing with both this flag and the 3rd parameter to this function having the same name but different types.. It would make it clearer if either this variable was renamed to something like "isForcedContext" or the parameter to something like "forceContextName"

Reply:
Sure - I've renamed both the argument and the flag